lib/Db/OldPasswordMapper.php

```diff
@@ -39,7 +39,7 @@ public function __construct(IDBConnection $db) {
 	 * @return OldPassword[]
 	 */
 	public function getOldPasswords($uid, $length, $excludeForceExpired = true) {
-		/* @var IQueryBuilder $qb */
+		/* @var OCP\DB\QueryBuilder\IQueryBuilder $qb */
```

Comment:
Please share the error which is fixed through this

Reply:
> Somehow phan seemed to stupid to understand that the use statement was "used" by the PHPdoc for the $qb variable.

Not a problem within phan - phan analysies the AST ( after sourcecode has been convered ). For performance reasons, the AST does not have inline comments for every machine code - it would make it too slow. 
Comments are available on the class // method level

Reply:
So in this case the change here is the "correct" solution?
Because PHP execution has no need for the ``use`` statement. It was only conventions of PHPdoc parsers that were parsing the ``use`` statement for convenience to work out what ``IQueryBuilder`` meant.

But in (unusual?) cases where there are many ``@var OCP\DB\QueryBuilder\IQueryBuilder`` PHPdoc throughout a class, but no actual real use of it, then there will be duplication of ``OCP\DB\QueryBuilder\`` in multiple PHPdoc blocks.